enrichments/trace/internal/elastic/span.go

```diff
@@ -300,6 +303,40 @@ func (s *spanEnrichmentContext) setServiceTarget(span ptrace.Span) {
 	}
 }
 
+func (s *spanEnrichmentContext) setDestinationService(span ptrace.Span) {
+	var destnResource string
+	if s.peerService != "" {
+		destnResource = s.peerService
+	}
```

Comment:
Or separate the derivation from setting the attribute, so we can return early.

Reply:
```go
		case s.isMessaging:
			if s.messagingSystem != "" {
				destnResource = s.messagingSystem
			}
			// For parity with apm-data, destn resource does not handle
			// temporary destination flag. However, it is handled by
			// service.target fields and we might want to do the same here.
			if s.messagingDestinationName != "" {
				destnResource += "/" + s.messagingDestinationName
			}
```

This is the problematic bit, the destination resource is updated based on the messagingDestinationName. I could make it return early with some conditions but, I think, that would make the code a bit harder to follow through and maintain. No strong opinions though but a simple if/else or a case in the switch statement itself for `s.peerService != ""` will not work.

Reply:
In case it was not clear in the above statement, the messaging destination is appended to the destination resource if it is not empty. Atleast, this is what parity with apm-data would be. This would mean that if `peer.service` is `testsvc` and `messagingDestinationName` is `topic1` then the destination resource would be `testsvc/topic1` -- this is also covered in the test case `TestElasticSpanEnrich/messaging_destination`

Reply:
Ah I was thinking we could do something like this:

```go
if s.messagingSystem != "" {
    destnResource := s.messagingSystem
    if s.messagingDestinationName != "" {
        destnResource += "/" + s.messagingDestinationName
    }
    return destnResource
}
```

I realise now that's not quite equivalent to what you've implemented though, because as it's written we might append the `messaging.destination.name` to `peer.service`. Is that a valid thing to do though?

Reply:
>  Is that a valid thing to do though?

I spent a bit of time reasoning about this when I was reading the apm-data code but I couldn't think of this NOT being valid so I carried over the logic. Technically, if we know the destination name of a topic and we have `peer.service` then appending the topic would result in an useful valid value. However, I am not sure how this would look practically for ex: will the `peer.service` and `messaging.destination.name` be populated together?

I am not too sure overall what to do here. I would suggest we do this separate to this PR, I have created a [GH issue](https://github.com/elastic/opentelemetry-lib/issues/74) to discuss this and make any changes to this. Let me know if it sounds good.

Reply:
Seems fine to me, thanks.